src/client-side-encryption/cryptoCallbacks.ts

```diff
@@ -1,11 +1,16 @@
 import * as crypto from 'crypto';
 
-export function makeAES256Hook(method, mode) {
-  return function (key, iv, input, output) {
+type AES256Callback = (key: Buffer, iv: Buffer, input: Buffer, output: Buffer) => number | Error;
+
+export function makeAES256Hook(
+  method: 'createCipheriv' | 'createDecipheriv',
+  mode: 'aes-256-cbc' | 'aes-256-ctr'
+): AES256Callback {
+  return function (key: Buffer, iv: Buffer, input: Buffer, output: Buffer): number | Error {
```

Comment:
I think we should strive to lower the contract needed here, the APIs accept Uint8Array as readily as they do Buffers. Are there Buffer specific APIs needed here?